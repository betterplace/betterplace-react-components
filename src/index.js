export * from './components/project_teaser'
export * from './components/share_buttons'
export * from './helpers/color_helper'
export * from './helpers/format_amount'
export * from './helpers/sharing_helper'
export * from './preview'
