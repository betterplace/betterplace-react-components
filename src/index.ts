export { ProjectTeaser } from './components/project_teaser'
export { FundraisingEventTeaser } from './components/fundraising_event_teaser'

// sharing
export { emailShareAction, EmailButton } from './components/share_buttons/email_button'
export { facebookShareAction, FacebookButton } from './components/share_buttons/facebook_button'
export {
  facebookMessengerShareAction,
  FacebookMessengerButton,
} from './components/share_buttons/facebook_messenger_button'
export { telegramShareAction, TelegramButton } from './components/share_buttons/telegram_button'
export { twitterShareAction, TwitterButton } from './components/share_buttons/twitter_button'
export { whatsappShareAction, WhatsappButton } from './components/share_buttons/whatsapp_button'
export { linkAction, LinkButton } from './components/share_buttons/link_button'
export { instagramAction, InstagramButton } from './components/share_buttons/instagram_button'

// helpers
export { changeBrightness } from './helpers/color_helper'
export { formatAmount } from './helpers/format_amount'
export { encodedShareURL, openPopup } from './helpers/sharing_helper'
