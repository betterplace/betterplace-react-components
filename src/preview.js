import React from 'react'
import { render } from 'react-dom'
import { ProjectTeaser } from './components/project_teaser'

const teaserNode = document.getElementById('brc-preview-teaser')

teaserNode &&
fetch('https://www.bp42.com/api_v4/projects/1114.json')
  .then(response => response.json())
  .then(project =>
    render(
      <ProjectTeaser
        project={project}
        showDescription={true}
        progressbarColor='#9ECB0A'
        textColor='#000000'
      />,
      teaserNode
    )
  )
