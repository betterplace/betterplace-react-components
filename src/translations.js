import I18n from 'i18n-js'
global.I18n = global.I18n || I18n

I18n.locale = document.documentElement.lang || 'de'
I18n.translations = I18n.translations || {}

// pass existing translations as second arg so that they are not overridden
I18n.translations = I18n.extend({
  de: {
    bp_project_teaser: {
      values_donor_count: "Spenden",
      values_open_amount_in_cents: "fehlen noch"
    }
  },
  en: {
    bp_project_teaser: {
      values_donor_count: "donations",
      values_open_amount_in_cents: "still needed"
    }
  }
}, I18n.translations)
