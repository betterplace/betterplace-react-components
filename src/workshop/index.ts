export * from './timer_widget';
