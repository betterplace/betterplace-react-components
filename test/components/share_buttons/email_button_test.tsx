import { shallow } from 'enzyme';
import React from 'react';

/* eslint-disable react/react-in-jsx-scope */
import { EmailButton } from '../../../src';

const shareUrl = 'https://foo.bar'
const shape = 'round'

describe('EmailButton', () => {
  it('renders a round button', () => {
    const element = shallow(<EmailButton shareUrl={shareUrl} shape={shape} />)
    expect(element).toMatchSnapshot()
  })

  it('does render if boxShadow is true', () => {
    const element = shallow(<EmailButton shareUrl={shareUrl} shape={shape} boxShadow />)
    expect(element).toMatchSnapshot()
  })

  it('does render if utmParams are provided', () => {
    const element = shallow(
      <EmailButton
        shareUrl={shareUrl}
        shape={shape}
        utmParams={{ utm_campaign: 'user_share', utm_medium: 'foo', utm_source: 'bar' }}
      />
    )
    expect(element).toMatchSnapshot()
  })

  it('adds a beforeClick function for the click handler', () => {
    const element = shallow(<EmailButton shareUrl={shareUrl} shape={shape} beforeOnClick={() => 'hey there'} />)
    expect(element).toMatchSnapshot()
  })
})
