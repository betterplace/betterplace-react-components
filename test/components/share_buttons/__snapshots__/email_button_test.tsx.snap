// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailButton adds a beforeClick function for the click handler 1`] = `
<ButtonRound
  aria-label="Email"
  buttonClassName="share-button--email"
  className=""
  href="#"
  icon={[Function]}
  label="Email"
  onClick={[Function]}
  role="button"
  title=""
/>
`;

exports[`EmailButton does render if boxShadow is true 1`] = `
<ButtonRound
  aria-label="Email"
  boxShadow={true}
  buttonClassName="share-button--email"
  className=""
  href="#"
  icon={[Function]}
  label="Email"
  onClick={[Function]}
  role="button"
  title=""
/>
`;

exports[`EmailButton does render if utmParams are provided 1`] = `
<ButtonRound
  aria-label="Email"
  buttonClassName="share-button--email"
  className=""
  href="#"
  icon={[Function]}
  label="Email"
  onClick={[Function]}
  role="button"
  title=""
/>
`;

exports[`EmailButton renders a round button 1`] = `
<ButtonRound
  aria-label="Email"
  buttonClassName="share-button--email"
  className=""
  href="#"
  icon={[Function]}
  label="Email"
  onClick={[Function]}
  role="button"
  title=""
/>
`;
